src/components/plot/bookmarkbutton.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import * as TetherComponent from 'react-tether';
+import {FacetedCompositeUnitSpec} from 'vega-lite/build/src/spec';
+import {BOOKMARK_ADD_PLOT, BOOKMARK_REMOVE_PLOT, BookmarkAction} from '../../actions/bookmark';
+import {ActionHandler} from '../../actions/redux-action';
+import {Bookmark} from '../../models/bookmark';
+import {PlotFieldInfo} from '../../models/plot';
+import * as styles from './bookmarkbutton.scss';
+
+
+
+export interface BookmarkProps extends ActionHandler<BookmarkAction> {
+  bookmark: Bookmark;
+  plotObjectFieldInfos: PlotFieldInfo[];
```

Comment:
If you really need both fieldInfo and spec, why don't you just pass the plotObject in here directly?